pydantic/main.py

```diff
@@ -957,12 +957,6 @@ def validate_model(  # noqa: C901 (ignore complexity)
             return {}, set(), ValidationError([ErrorWrapper(exc, loc=ROOT_KEY)], cls_)
 
     for name, field in model.__fields__.items():
-        if field.type_.__class__ == ForwardRef:
```

Comment:
I thought putting this logic in `validate` was easier but we could also change this line into
```python
if ForwardRef in {t.__class__ for t in get_all_args(field.type_)}:
```

where 

```python
def get_all_args(type_: Any) -> Set[type]:
    from .typing import get_args

    args = get_args(type_)
    types: Set[type] = set()

    if args:
        for t in args:
            types.update(get_all_args(t))
    else:
        types.add(type_)

    return types
```